### PR TITLE
chore: moving conedison/provider to interface

### PR DESCRIPTION
**Pull request by JackShort**
* moves conedison provider utilis to interface as we are deprecating conedison
* lateral pr to #7114 
* removes conedison
* updates jest types to v27 from v25 to support `lastCall`
* removes WalletConnectV1 code from conedison signing functions

### CODE REVIEW

File: craco.config.cjs [hunk 1]
```diff
@@ -56,10 +56,10 @@ module.exports = {
           '\\.css\\.ts$': '@vanilla-extract/jest-transform',
           '\\.(t|j)sx?$': '@swc/jest',
         },
-        // Use @uniswap/conedison's build directly, as jest does not support its exports.
-        transformIgnorePatterns: ['@uniswap/conedison/provider'],
+        // Use d3-arrays's build directly, as jest does not support its exports.
+        transformIgnorePatterns: ['d3-array'],
         moduleNameMapper: {
-          '@uniswap/conedison/provider': '@uniswap/conedison/dist/provider',
+          'd3-array': 'd3-array/dist/d3-array.min.js',
```
Comment (zzmp) [inline craco.config.cjs:62]: Why did you add `d3-array` in here? Does it _need_ to be here?

**Review by zzmp — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -84,7 +84,7 @@
     "@types/array.prototype.flat": "^1.2.1",
     "@types/array.prototype.flatmap": "^1.2.2",
     "@types/d3": "^6.7.1",
-    "@types/jest": "^25.2.1",
+    "@types/jest": "^27.0.1",
```
[the remark this answers is not in the document]
Comment (JackShort) [reply package.json]: sorry - it didn't seem to have a problem when testing it was just to add a lastCall type to it but i can just get it through the `calls` array

File: craco.config.cjs (hunk 1 above)
Comment (JackShort) [reply craco.config.cjs:62]: yes it needs to be here. jest doesn't like d3-arrays exports and for some reason using conedison's build applied to d3-array, but when removed it needs its own

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (zzmp) [reply package.json]: if the typing is accurate then this is great!

**Review by zzmp — APPROVED** [review]
i think you should upgrade the jest typing :)

**Review by zzmp — APPROVED** [review]

**Review by zzmp — APPROVED** [review]
